### PR TITLE
fix(ledger): fail closed on committee certificates

### LINKED ISSUE #2142: ledger: fail closed on missing constitutional committee state
## Summary

Conway committee certificate authorization currently treats an empty or unreadable committee set as if membership checks do not apply. Authorization must fail closed when the committee state is empty or unavailable.

## Acceptance criteria

- Reject unauthorized hot-key authorization and cold-key resignation when the committee set is empty.
- Propagate committee-state lookup errors instead of converting them into authorization success.
- Consult proposed committee members consistently with the ledger semantics.
- Preserve valid seated-member and proposal-member authorization.
- Add regression tests for seated, empty, lookup-error, and proposed-member states.
- Run full normal/race, conformance, lint, vet, and diff checks.

## Scope

Keep the change within Conway committee certificate validation and its ledger-state contract.

### PR DESCRIPTION
**Pull request by wolf31o2**
Fixes #2142.

- preserves exact key and script committee credentials
- fails closed on unavailable, empty, or inconsistent committee state
- validates authorization, resignation, and committee voters against active membership
- reports committee voter lookup failures as `CommitteeMemberLookupError`, carrying the credential and its type rather than a bare provider error
- skips governance checks for phase-2-invalid transactions
- preserves released and exact-provider conformance coverage

The legacy conformance adapter reports a script credential as not a member: the released hash-only store cannot represent one, so answering it could only alias onto the key-hash entry of the same hash.

### CODE REVIEW

File: internal/test/conformance/conformance_test.go [hunk 1]
```diff
@@ -17,7 +17,132 @@ package conformance
...
+func (p *legacyCommitteeStateProvider) CommitteeStateAvailable() (bool, error) {
+	// MockLedgerState always initializes committee state, including the
+	// authoritative-empty case.
+	return true, nil
+}
+
+func (p *legacyCommitteeStateProvider) CommitteeCredentialMember(
+	credential common.Credential,
+) (*common.CommitteeMember, error) {
+	member, err := p.CommitteeMember(credential.Credential)
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply internal/test/conformance/conformance_test.go:79]: Valid. Fixed in 03d2d53 by reporting not a member for any credential whose `CredType` is not `CredentialTypeAddrKeyHash`, in both `CommitteeCredentialMember` and `CommitteeHotCredentialMember`.

The released v0.17.0 store is keyed by hash alone and cannot represent a script member, so answering a script credential from it could only ever alias onto the key-hash entry with the same hash. Returning nil states that limitation instead of masking it, which is the contract `CommitteeCredentialState` is there to hold.

All 315 conformance vectors still pass, so no vector depended on the aliasing behaviour.

File: ledger/conway/rules.go [hunk 2]
```diff
@@ -3523,38 +3583,34 @@ func UtxoValidateUnknownVoters(
...
-				// member's hot key is treated as a valid voter here. This
-				// is the same class of LedgerState-surface limitation
-				// disclosed via NOTE comments elsewhere in this file
-				// (e.g. UtxoValidateProposalAncestry, UtxoValidateHardForkCanFollow).
-				if member.HotKey != nil && *member.HotKey == hotHash &&
-					!member.Resigned {
-					found = true
-					break
-				}
+			member, err := committeeState.CommitteeHotCredentialMember(
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply ledger/conway/rules.go:3615]: Valid. Fixed in 03d2d53. The three raw returns in the committee voter branch of `UtxoValidateUnknownVoters` (`CommitteeStateUnavailableError`, the `CommitteeStateAvailable` error, and the `CommitteeHotCredentialMember` error) now go through a `CommitteeMemberLookupError`, matching `UtxoValidateCommitteeCertificates`.

Since a bare `Blake2b224` cannot distinguish a hot key hash from a hot script hash, and that is the same aliasing raised on the conformance adapter, `CommitteeMemberLookupError.ColdCredential` is renamed to `MemberCredential`: the cold credential for a certificate lookup, the hot credential for a voter lookup. The field was added by this PR, so nothing outside it referenced the old name.

`TestUtxoValidateUnknownVotersWrapsCommitteeLookupFailures` covers both voter types against a state with no committee capability and against a failing provider. All four subtests fail on the previous code.

**Review by arepala-uml — APPROVED** [review]
Verified independently against the current checkout (build, vet, `make lint`, `nilaway`, `go test -race ./ledger/...`, and `go test ./internal/test/conformance/...`): all pass, 315/315 conformance vectors pass. CodeRabbit and Cubic have no actionable findings; all CI checks are green.

Traced the new `common.CommitteeCredentialState` capability across the dependency spine: `ouroboros-mock`'s shared `MockLedgerState` already implements it (merged in ouroboros-mock#263), and Dingo's `LedgerView` has a matching structural implementation in progress (dingo#3745, open). Release sequencing note: don't let Dingo bump to a gouroboros release containing this fix before dingo#3745 merges, or committee certificates/CC votes will fail closed in production until that lands.

No merge-blocking findings.

File: ledger/conway/rules.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (wolf31o2) [reply ledger/conway/rules.go:3615]: Not valid against the merged head 03d2d53c. Every committee lookup in
`UtxoValidateUnknownVoters` already wraps through the local `lookupError`
helper, which returns `CommitteeMemberLookupError` carrying the credential:

- 3605 unavailable state
- 3609 `CommitteeStateAvailable` failure
- 3612 unavailable after the check
- 3619 `CommitteeHotCredentialMember` failure

The one raw `return err`, at 3573, is the `ls.DRepRegistration` path. That is a
DRep credential, not a committee credential, so wrapping it in
`CommitteeMemberLookupError` would attach a committee member shape to a
non-committee lookup and misreport the source of the failure. Leaving it
unwrapped is deliberate, not an inconsistency with
`UtxoValidateCommitteeCertificates`.

File: internal/test/conformance/conformance_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wolf31o2) [reply internal/test/conformance/conformance_test.go:79]: Not valid against the merged head 03d2d53c; the adapter does the opposite of
what this describes.

Both `CommitteeCredentialMember` and `CommitteeHotCredentialMember` open by
rejecting any non-key credential before any hash lookup happens:

    if credential.CredType != common.CredentialTypeAddrKeyHash {
        return nil, nil
    }

So a script cold or hot credential is reported as not a member rather than
aliased onto a key-hash member sharing its hash. `CredType` is read, not
ignored, and the in-code comment names that aliasing as the reason. The
hash-only lookups that follow are reachable only for
`CredentialTypeAddrKeyHash`, where keying on the hash alone is correct because
the released v0.17.0 store this fallback adapts cannot represent a script
member at all.
